java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -254,38 +303,84 @@ public static Table readCSV(Schema schema, CSVOptions opts, HostMemoryBuffer buf
         opts.getFalseValues()));
   }
 
+  /**
+   * Read a Parquet file using the default ParquetOptions.
+   * @param path the local file to read.
+   * @return the file parsed as a table on the GPU.
+   */
   public static Table readParquet(File path) {
     return readParquet(ParquetOptions.DEFAULT, path);
   }
 
+  /**
+   * Read a Parquet file.
+   * @param opts various parquet parsing options.
+   * @param path the local file to read.
+   * @return the file parsed as a table on the GPU.
+   */
   public static Table readParquet(ParquetOptions opts, File path) {
     return new Table(gdfReadParquet(opts.getIncludeColumnNames(),
         path.getAbsolutePath(), 0, 0));
   }
 
+  /**
+   * Read parquet formatted data.
+   * @param buffer raw parquet formatted bytes.
+   * @return the data parsed as a table on the GPU.
+   */
   public static Table readParquet(byte[] buffer) {
-    return readParquet(ParquetOptions.DEFAULT, buffer, buffer.length);
+    return readParquet(ParquetOptions.DEFAULT, buffer, 0, buffer.length);
   }
 
+  /**
+   * Read parquet formatted data.
+   * @param opts various parquet parsing options.
+   * @param buffer raw parquet formatted bytes.
+   * @return the data parsed as a table on the GPU.
+   */
   public static Table readParquet(ParquetOptions opts, byte[] buffer) {
-    return readParquet(opts, buffer, buffer.length);
+    return readParquet(opts, buffer, 0, buffer.length);
   }
 
-  public static Table readParquet(ParquetOptions opts, byte[] buffer, long len) {
+  /**
+   * Read parquet formatted data.
+   * @param opts various parquet parsing options.
+   * @param buffer raw parquet formatted bytes.
+   * @param offset the starting offset into buffer.
+   * @param len the number of bytes to parse.
+   * @return the data parsed as a table on the GPU.
+   */
+  public static Table readParquet(ParquetOptions opts, byte[] buffer, long offset, long len) {
     if (len <= 0) {
-      len = buffer.length;
+      len = buffer.length - offset;
     }
     assert len > 0;
-    assert len <= buffer.length;
+    assert len <= buffer.length - offset;
+    assert offset >= 0 && offset < buffer.length;
     try (HostMemoryBuffer newBuf = HostMemoryBuffer.allocate(len)) {
       newBuf.setBytes(0, buffer, 0, len);
-      return readParquet(opts, newBuf, len);
+      return readParquet(opts, newBuf, 0, len);
```

Comment:
I think this shouldn't be 0 and offset should be factored in.

Reply:
great catch.

Reply:
@abellina please take another look

Reply:
Should we add a test that passes an offset instead of 0 as an offset

Reply:
That is a lot harder to do with Parquet than it is with CSV.  So I would prefer not to, but if you really want one I can probably put one together.

Reply:
Thanks @revans2. Would a simple test like allocate a slightly bigger byte[] than some Parquet buffer, then left pad it with something, s.t. we just have this read the Parquet payload at the offset?

Reply:
I am OK with it if you are OK